hazelcast/include/hazelcast/client/serialization/pimpl/compact.i.h

```diff
@@ -220,6 +220,88 @@ compact_reader::read()
     return values;
 }
 
+template<typename T>
+typename std::enable_if<
+  std::is_same<big_decimal, typename std::remove_cv<T>::type>::value,
+  typename boost::optional<T>>::type
+compact_reader::read()
```

Comment:
I suggest not to put it in `object_data_input` because for example time/date are different between portable and compact. 
I can put to an IOUtil to be share the code between BuiltinSerializers and Compact as in java. I was postponing that work to be done when sharing actually is needed.